docs/expand.md

```diff
@@ -0,0 +1,82 @@
+# Expand
+
+## Contents
+
+1. [Overview](#overview)
+2. [Specifying expansions](#specifying-expansions)
+3. [Annotating expansion results](#annotating-expansion-results)
+
+## Overview
+
+Some logical patterns in OWL are cumbersome to maintain by hand; shortuct annotation properties ("macros") can be used as an
+editor's representation which are automatically expanded to more complex axioms.
+
+For example, an ontology may include the notion of "part disjointness", i.e., nothing with a 'part of' relationship
+to an instance of concept A has a 'part of' relationship to any instance of concept B. This can be represented using
+a GCI like so:
+
+```
+('part of' some A) DisjointWith ('part of' some B)
+```
+
+Alternatively, one could define an annotation property `'part disjoint with'`:
+
+```
+A 'part disjoint with' B
+```
+
+This property definition can be annotated with another property, [defined by construct](http://purl.obolibrary.org/obo/OMO_defined_by_construct),
+pointing to a SPARQL CONSTRUCT query, which is used to generate the more complex OWL axioms when running `robot expand`:
+
+```
+'part disjoint with' 'defined by construct' """
+    PREFIX owl: <http://www.w3.org/2002/07/owl#>
+    PREFIX : <http://example.org/
+    CONSTRUCT {
+      [
+        a owl:Restriction ;
+        owl:onProperty :part_of ;
+        owl:someValuesFrom ?a ;
+        owl:disjointWith [
+          a owl:Restriction ;
+          owl:onProperty :part_of ;
+          owl:someValuesFrom ?b
+        ]
+      ]
+    }
+    WHERE {
+      ?a :part_disjoint_with ?b .
+    }
+    """
+```
+
+Here is an example:
+
+    robot expand \
+        --input unexpanded.ttl \
+        --output results/expanded.ttl
+
+## Specifying expansions
+
+By default, all SPARQL CONSTRUCT queries found as values for the [defined by construct](http://purl.obolibrary.org/obo/OMO_defined_by_construct) property
+will be executed against the input ontology, and the results merged in.
+
+Alternatively, specific macro terms can be included or excluded from the expansion process:
+
+    robot expand \
+        --input unexpanded.ttl \
+        --expand-term 'http://example.org/never_part_of' \
```

Comment:
Can I expect --expand-term options to be executed in order of how we specify them in the command? What is the default order if no expand-term are given?

Reply:
Order is irrelevant. The results are not allowed to interact.

Reply:
Ok, so if I have expansions that interact, I can always do 

`expand --expand-term A expand --expand-term B`

Thanks! Mayne add a comment to that end to the docs (that order is irrelevant).

Reply:
Okay, I did that.

Reply:
I haven't had a chance to review yet, sorry. Here's a thought (not a blocker or a requirement):

Because of my short-sighted design decision years ago, ROBOT only passes an `OWLOntology` between chained commands. So for a chain of expand commands like this, we are repeatedly calling `loadOntologyAsDataset()`. That's inefficient. If that's an important use case to support, or if we're worried about handling very large ontologies, maybe we could borrow from <http://robot.obolibrary.org/query#executing-on-disk>. We could add support for this later as long as there's room in the design.

So right now I'm just asking a question: Is there room to add TDB options later?

Reply:
@jamesaoverton it would be really easy to add a TDB option - you would just need to use this (maybe this should be moved to `QueryOperation` if we're going to reuse it.): https://github.com/ontodev/robot/blob/00ad6b8bf901757c3548ae18aa82ddcf8623a815/robot-command/src/main/java/org/obolibrary/robot/QueryCommand.java#L198

... instead this: 
https://github.com/ontodev/robot/blob/2efe4614357eb90c41b45abd92b2802359b1d020/robot-core/src/main/java/org/obolibrary/robot/ExpandOperation.java#L93

... and then you can execute everything on the `Dataset` the same way.